deep_qa/data/instances/mc_tuples_with_background_instance.py

```diff
@@ -0,0 +1,358 @@
+from collections import defaultdict
+from typing import Dict, List
+
+import numpy as np
+from overrides import overrides
+
+from .instance import TextInstance, IndexedInstance
+from ..data_indexer import DataIndexer
+
+
+class TextTuple:
+    def __init__(self, subject: str, verb: str, objects: List[str],
+                 context: str=None, time: str=None, loc: str=None, source: str=None):
+        self.subject = subject
+        self.verb = verb
+        self.objects = objects
+        self.context = context
+        self.time = time
+        self.location = loc
+        self.source = source
+
+    def __str__(self):
+        # NOTE: not currently displaying locaton, time, and source info
+        return "TextTuple( [\n\tsubj: {0} \n\tverb: {1} \n\tobjects: {2} \n\tcontext: {3}\n] )"
+
+    def to_text_list(self, object_handling: str="collapse", include_context: bool=True):
+        # Subject and verb
+        text_list = [self.subject, self.verb]
+        # Object(s)
+        if object_handling == "collapse":
+            text_list.append(" ".join(self.objects))
+        elif object_handling == "first":
+            if len(self.objects) > 0:
+                text_list.append(self.objects[0])
+        else:
+            raise Exception("Requested object_handling ({0}) not currently supported.".format(object_handling))
+        # Context
+        if include_context and self.context:
+            text_list.append(self.context)
+        # Currently, source, time and location aren't being used, but this can be changed in the future.
+
+        return text_list
+
+
+    @classmethod
+    def from_string(cls, input_string: str):
+        tuple_fields = input_string.lower().split("|")
```

Comment:
I will msg tushar about changing it, but until he changes his end, I am hesitant to change mine, as I need to maintain compatibility.